internal/controller/apiextensions/composite/reconciler.go

```diff
@@ -379,27 +401,15 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 	}
 	r.record.Event(cr, event.Normal(reasonResolve, "Successfully selected composition"))
 
-	// TODO(muvaf): We should lock the deletion of Composition via finalizer
-	// because its deletion will break the field propagation.
-	comp := &v1.Composition{}
-	if err := r.client.Get(ctx, meta.NamespacedNameOf(cr.GetCompositionReference()), comp); err != nil {
-		log.Debug(errGetComp, "error", err)
-		r.record.Event(cr, event.Warning(reasonCompose, err))
-		return reconcile.Result{RequeueAfter: shortWait}, nil
-	}
-
-	if err := r.composite.Configure(ctx, cr, comp); err != nil {
-		log.Debug(errConfigure, "error", err)
+	// Note that this 'Composition' will be derived from a
+	// CompositionRevision if the relevant feature flag is enabled.
+	comp, err := r.composition.Fetch(ctx, cr)
+	if err != nil {
+		log.Debug(errFetchComp, "error", err)
 		r.record.Event(cr, event.Warning(reasonCompose, err))
 		return reconcile.Result{RequeueAfter: shortWait}, nil
 	}
 
-	log = log.WithValues(
-		"composition-uid", comp.GetUID(),
-		"composition-version", comp.GetResourceVersion(),
-		"composition-name", comp.GetName(),
-	)
```

Comment:
This was removed because if we derive our composition from a revision it won't actually have a UID, version, or name.